core/src/main/java/com/linecorp/armeria/server/ServiceRequestContextBuilder.java

```diff
@@ -234,7 +244,9 @@ public ServiceRequestContext build() {
         }
 
         final CancellationScheduler requestCancellationScheduler;
-        if (timedOut()) {
+        if (!useCancellationScheduler) {
+            requestCancellationScheduler = CancellationScheduler.noop();
+        } else if (timedOut()) {
             requestCancellationScheduler = CancellationScheduler.finished(true);
```

Comment:
Is there a way to create a context that's been timed out without submitting any tasks? Then we could allow our users to create such a context without performance penalty. This could be done as a follow-up task, though.